### PR TITLE
qt5-build.eclass: Add quirks

### PR DESCRIPTION
**Pull request by a17r**

### CODE REVIEW

File: eclass/qt5-build.eclass [hunk 1]
```diff
@@ -326,6 +326,23 @@ qt5-build_pkg_postrm() {
 
 ######  Public helpers  ######
 
+# @FUNCTION: qt5_syncqt_version
+# @DESCRIPTION:
+# Wrapper for Qt5 syncqt.pl to sync header files for ${PV} (required to run if
+# headers are added/removed by patching
```
Comment (Pesa) [inline eclass/qt5-build.eclass:332]: missing `)`

File: eclass/qt5-build.eclass [hunk 2]
```diff
@@ -720,11 +720,37 @@ qt5_tools_configure() {
...
+			# src/plugins/geoservices requires files that are only generated
+			# when qmake is run in the root directory. bug 633776
+			qt5_tools_oos_quirk src/location qtlocation-config.pri
+			;;
+		*) ;;
+	esac
+
 	mkdir -p "${QT5_BUILD_DIR}" || die
 	qt5_qmake "${QT5_BUILD_DIR}"
 	cp qttools-config.pri "${QT5_BUILD_DIR}" || die
```
Comment (Pesa) [inline eclass/qt5-build.eclass:756]: this code looks fairly similar, can we use `qt5_tools_oos_quirk` here as well? (might need to swap the arguments and make the directory optional, or make the logic in `qt5_tools_oos_quirk` a little smarter if only one argument is passed)

**Review by Pesa — APPROVED** [review]

File: eclass/qt5-build.eclass [hunk 3]
```diff
@@ -720,11 +737,37 @@ qt5_tools_configure() {
...
 }
 
+# @FUNCTION: qt5_tools_oos_quirk
+# @INTERNAL
+# @USAGE: <path> <file>
+# @DESCRIPTION:
+# Runs qmake in root directory, copies generated pri <file> from source <path>
```
Comment (thesamesam) [inline eclass/qt5-build.eclass:763]: Sorry to be annoying, but oos may not be obvious to everyone (I know chances of others reading it are small anyway, but well):
```suggestion
# Quirk for out-of-source builds. Runs qmake in root directory, copies generated pri <file> from source <path>
```

File: dev-qt/qtcore/qtcore-5.15.10.9999.ebuild [hunk 4]
```diff
@@ -79,6 +79,11 @@ src_prepare() {
 	fi
 
 	qt5-build_src_prepare
+
+	if [[ ${PV} != *9999* ]]; then
```
Comment (thesamesam) [inline dev-qt/qtcore/qtcore-5.15.10.9999.ebuild:83]: Should we just make the function a noop for 9999?

Comment (a17r) [reply dev-qt/qtcore/qtcore-5.15.10.9999.ebuild:83]: Makes sense, I guess.

File: eclass/qt5-build.eclass (hunk 3 above)
Comment (a17r) [reply eclass/qt5-build.eclass:763]: That's what I'm here for.

**Review by thesamesam — APPROVED** [review]

**Review by Pesa — APPROVED** [review]

**Pull request merged**